tests/fast_int.cpp

```diff
@@ -0,0 +1,565 @@
+    std::cerr << "ptr of result "  << std::quoted(f3) << " did not match with expected ptr: " << std::quoted(" with text") << std::endl;
+    return EXIT_FAILURE;
+  }
+
+  // int pointer test #4 (float)
+  const std::string int_pointer_test_4 = "9.999";
+
+  const auto& f4 = int_pointer_test_4;
+  int result4;
+  auto answer4 = fast_float::from_chars(f4.data(), f4.data() + f4.size(), result4);
+  if (strcmp(answer4.ptr, ".999") != 0) {
+    std::cerr << "ptr of result "  << std::quoted(f4) << " did not match with expected ptr: " << std::quoted(".999") << std::endl;
+    return EXIT_FAILURE;
+  }
+
+  // int pointer test #5 (invalid int)
+  const std::string int_pointer_test_5 = "+50";
+
+  const auto& f5 = int_pointer_test_5;
+  int result5;
+  auto answer5 = fast_float::from_chars(f5.data(), f5.data() + f5.size(), result5);
+  if (strcmp(answer5.ptr, "+50") != 0) {
+    std::cerr << "ptr of result "  << std::quoted(f5) << " did not match with expected ptr: " << std::quoted("+50") << std::endl;
+    return EXIT_FAILURE;
+  }
+
+  // unsigned pointer test #2 (string behind numbers)
+  const std::string unsigned_pointer_test_1 = "1001 with text";
+
+  const auto& f6 = unsigned_pointer_test_1;
+  unsigned result6;
+  auto answer6 = fast_float::from_chars(f6.data(), f6.data() + f6.size(), result6);
+  if (strcmp(answer6.ptr, " with text") != 0) {
+    std::cerr << "ptr of result "  << std::quoted(f6) << " did not match with expected ptr: " << std::quoted(" with text") << std::endl;
+    return EXIT_FAILURE;
+  }
+
+  // unsigned pointer test #2 (invalid unsigned)
+  const std::string unsigned_pointer_test_2 = "-50";
+
+  const auto& f7 = unsigned_pointer_test_2;
+  unsigned result7;
+  auto answer7 = fast_float::from_chars(f7.data(), f7.data() + f7.size(), result7);
+  if (strcmp(answer7.ptr, "-50") != 0) {
+    std::cerr << "ptr of result "  << std::quoted(f7) << " did not match with expected ptr: " << std::quoted("-50") << std::endl;
+    return EXIT_FAILURE;
+  }
+
+  // int base 2 test
+  const std::vector<int> int_base_2_test_expected { 0, 1, 4, 2, -1 };
+  const std::vector<std::string> int_base_2_test { "0", "1", "100", "010", "-1" };
+
+  for (std::size_t i = 0; i < int_base_2_test.size(); ++i)
+  {
+    const auto& f = int_base_2_test[i];
+    int result;
+    auto answer = fast_float::from_chars(f.data(), f.data() + f.size(), result, 2);
+    if (answer.ec != std::errc()) {
+      std::cerr << "could not convert to int for input: " << std::quoted(f) << std::endl;
+      return EXIT_FAILURE;
+    }
+    else if (result != int_base_2_test_expected[i]) {
+      std::cerr << "result "  << std::quoted(f) << " did not match with expected int: " << int_base_2_test_expected[i] << std::endl;
+      return EXIT_FAILURE;
+    }
+  }
+
+  // unsigned base 2 test
+  const std::vector<unsigned> unsigned_base_2_test_expected { 0, 1, 4, 2 };
+  const std::vector<std::string> unsigned_base_2_test { "0", "1", "100", "010" };
```

Comment:
for every valid test, can you also add cases where the numbers are just within range (like int_min or int_max)